support-frontend/assets/helpers/stripe.js

```diff
@@ -51,9 +38,47 @@ function getStripeKey(stripeAccount: StripeAccount, country: IsoCountry, isTestU
         window.guardian.stripeKeyDefaultCurrencies[stripeAccount].default;
   }
 }
+//  this is required as useStripeObjects is used in multiple components
+//  but we only want to call setLoadParameters once.
+const stripeScriptHasBeenAddedToPage = (): boolean =>
+  !!document.querySelector('script[src^=\'https://js.stripe.com\']');
+
+export const useStripeObjects = (stripeAccount: StripeAccount, stripeKey: string) => {
+  const [stripeObjects, setStripeObjects] = useState<{[StripeAccount]: StripeSDK | null}>({
+    REGULAR: null,
+    ONE_OFF: null,
+  });
+  useEffect(
+    () => {
+      if (stripeObjects[stripeAccount] === null) {
+        new Promise((resolve) => {
+          onConsentChange(({ ccpa, tcfv2 }) => {
+            if (ccpa) {
+              resolve(true);
+            }
+            if (tcfv2 && tcfv2.consents[1]) {
+              resolve(true);
+            }
+            resolve(false);
+          });
+        }).then((hasRequiredConsentsForFraudDetection) => {
+          if (!stripeScriptHasBeenAddedToPage()) {
+            loadStripe.setLoadParameters({ advancedFraudSignals: hasRequiredConsentsForFraudDetection });
+          }
+          return loadStripe(stripeKey);
+        }).then((newStripe) => {
+          setStripeObjects(prevData => ({ ...prevData, [stripeAccount]: newStripe }));
+        });
+      }
+    },
+    [stripeAccount],
+  );
+
+  return stripeObjects;
+};
+
 
 export {
-  setupStripe,
```

Comment:
Are we sure only the contributions checkout was dependent on this export?

Reply:
Yeah it was just being used in the `StripeCardFormContainer` and `StripePaymentRequestButtonContainer` components, which now both use `useStripeObjects` instead